### PR TITLE
Add Authentication to S3 API

### PR DESCRIPTION
**Pull request by bradyoo**

### CODE REVIEW

**Comment by bradyoo** [comment]
Tests will be added once https://github.com/Alluxio/alluxio/pull/12701 gets in. Tested with S3Browser for now.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java [hunk 1]
```diff
@@ -124,6 +124,18 @@ private static String getUserFromAuthorization(String authorization) {
     return stripped.substring(0, colonIndex);
   }
 
+  private FileSystem getFileSystem(String authorization) {
+    final String user = getUserFromAuthorization(authorization);
+
+    if (user == null) {
```
Comment (gpang) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:130]: `getUserFromAuthorization()` sometimes returns `null` or `""` (empty string). Should `""` be handled here as well?

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java [hunk 2]
```diff
@@ -701,40 +757,6 @@ private void checkUploadId(AlluxioURI multipartTemporaryDir, long uploadId) thro
     }
   }
 
-  private List<URIStatus> listObjects(AlluxioURI uri, ListBucketOptions listBucketOptions)
```
Comment (gpang) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:704]: why is this removed?

**Review by gpang — COMMENTED** [review]
@bradyoo Thanks! Just a few minor questions.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java (hunk 2 above)
Comment (bradyoo) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:704]: It's an unused code that became unused after previous commit when list Bucket was fixed.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java (hunk 1 above)
Comment (bradyoo) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:130]: Let me change them all to return null.

**Comment by gpang** [comment]
@bradyoo 
test failed:
```
Error: 5.012 [ERROR] alluxio.client.rest.S3ClientRestApiTest.listAllMyBuckets(alluxio.client.rest.S3ClientRestApiTest)
Error: 5.012 [ERROR]   Run 1: S3ClientRestApiTest.listAllMyBuckets:130 Request failed: <html>
```

**Comment by bradyoo** [comment]
Hmm I have tried to make an attempt all day on Friday to try to resolve this and getting nowhere. In particular, I can't seem to get the FileSystem on proxy to talk correctly for different users in test.

**Comment by bradyoo** [comment]
I think the issue has now been addressed.

**Comment by gpang** [comment]
@bradyoo 
```
/usr/src/alluxio/tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java:15:15: error: Unused import - org.mockito.ArgumentMatchers.any.
/usr/src/alluxio/tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java:127: error: Line is longer than 100 characters (found 107).
```

**Comment by bradyoo** [comment]
oops. addressed.

**Review by gpang — APPROVED** [review]
LGTM

**Comment by gpang** [comment]
alluxio-bot, merge this please
